[GH-618]: Added slack attachment for webhook posts, in order to comment, edit, and close issues.

Nityanand13 opened the pull request:
<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
<!--
A description of what this pull request does.
-->
Whenever a user creates an issue on github then three actions will be shown in the custom post created by the plugin's bot. The three actions are: 
- Comment
- Edit
- Close

With the help of the above mentioned actions user can create a comment, edit, close, and reopen a github issue from the mattermost. As you see we haven't made a reopen button so when a user closes an issue then this close button will turn to reopen.

![Screenshot from 2024-07-22 18-05-32](https://github.com/user-attachments/assets/e2239bf8-da22-4d56-a004-5ec9417e929b)

#### Ticket Link
<!--
If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.

  Fixes https://github.com/mattermost/mattermost-server/issues/XXXXX

Otherwise, link the JIRA ticket.
-->
Issue: #618

Nityanand13 committed:
[MI-2502]:Added custom type for webhook post in order to comment, edit and close issues (#13)

* [MI-2502]:Added custom type for webhook post in order to comment, edit and close issues

* [MI-2502]: Added EOF

* [MI-2502]: Fixed lint errors

* [MI-2502]: Review fixes done
1. Improved code quality
2. Made few contants

* [MI-2502]: Review fixes done
1. Changed few files to .tsx
2. Improved code quality

* [MI-2502]: Review fixes done
1. Improved code quality
2. Made a constant

* [MI-2502]: Review fixes done
1. Improved code readability

* [MI-2502]: Review fixes done
1. Made a package dor constants
2. Made a package for struct
3. Improved code redability

* [MI-2502]: Review fixes done
1. Improved code quality
2. Fixed the issue that I was facing while creating issue or attaching a comment to the issue.
3. Changed few file names

* [MI-2502]: Review fixes done
1. Changed the names of few functions
2. Improved code quality

* [MI-2502]: Review fixes done
1.Improved code readability

* [MI-2502]: Review fixes done
1. Improved code readability

* [MI-2502]: Review done
1. Improved code quality

* [MI-2502]: Review fixes done
1. Improved code quality

Nityanand13 committed:
Merge branch 'master' of github.com:mattermost/mattermost-plugin-github into MM-618

mattermost-build commented:
Hello @Nityanand13,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

hanzei reviewed (dismissed):
That is a very nifty feature :tada: 

One broad question before I dive into the code too much: The implementation currently uses a custom post and a webapp model. Could this feature also be implemented using interactive dialogues and post actions? That way it's supported on mobile.

hanzei commented:
Using the `Comment` feature posts this
![Screenshot from 2023-02-03 13-30-01](https://user-images.githubusercontent.com/16541325/216604142-35484a07-388d-4584-9028-ae807bd49084.png) message in the thread, which is misleading as no message from mattermost got attached.

Nityanand13 commented:
> That is a very nifty feature tada
> 
> One broad question before I dive into the code too much: The implementation currently uses a custom post and a webapp model. Could this feature also be implemented using interactive dialogues and post actions? That way it's supported on mobile.

@hanzei I think we can implement this feature using interactive dialogues because in that we do not have any case in which we can select more than one option for a particular field. 
For here if we want to edit an issue then there are fields like labels in which there might be a case in which we need to select more than one option from the list.

mickmister commented on webapp/src/components/modals/create_update_issue/create_update_issue.jsx:
```diff
@@ -122,24 +155,25 @@ export default class CreateIssueModal extends PureComponent {
             return null;
         }
 
+        const repoName = this.state.repo.name ?? this.state.repo;
```
Why is `this.state.repo` sometimes an object, and sometimes a string? Is it ever not defined?

Nityanand13 replied:
When we are selecting a repo while creating an issue then `this.state.repo` is storing an object and when we are editing an issue then the repository is fixed so in that case `this.state.repo` is storing a string.

mickmister replied:
@Nityanand13 Can we make it so we consistently store one data type, rather than making this state variable hold two different potential data types?

Nityanand13 replied:
Yes, we can do that.

Nityanand13 commented on webapp/src/components/modals/attach_comment_to_issue/attach_comment_to_issue.jsx:
```diff
@@ -83,29 +107,39 @@ export default class AttachIssueModal extends PureComponent {
+                    value={issueValue}
                 />
                 <Input
                     label='Message Attached to GitHub Issue'
                     type='textarea'
                     isDisabled={true}
-                    value={this.props.post.message}
+                    value={post?.message}
                     disabled={false}
                     readOnly={true}
                 />
```
In that case, we are getting a message something like `Please provide a valid non empty comment.`

![Screenshot from 2023-02-08 12-07-59](https://user-images.githubusercontent.com/107465508/217454172-685d68ef-e613-4105-bfd0-764e5a38bfa4.png)

mickmister replied:
Let's just remove the box if there are no contents and the user isn't going to type anything there. Why is the box blank anyway? What case do we not have a post? Maybe we should let the user type in a message regardless?

mickmister replied:
If the feature is to attach a comment, shouldn't we allow the user to type in a comment?

Nityanand13 replied:
In this PR I only fixed issue #618. Where if a user clicks on the comment button as shown in the image attached to the description of this PR. So a modal will open as shown below where a user can type a comment and attach that to the github issue.

![Screenshot from 2023-02-10 13-15-54](https://user-images.githubusercontent.com/107465508/218032960-a37b586d-c6fa-428c-8884-efcb0fd03f6d.png)

Other than this a user can create a github issue or attach a message to the github issue from message actions. For now, I have not changed the logic of this thing and the image of the modal that you are seeing where the user cannot type anything is getting opened when a user wants to attach a message to the github issue from message action. So in this case the message field usually contains the content of that particular post. And in the case of slack attachment posts, this message field remains empty. So should I make this field editable or what do you say?

Nityanand13 replied:
@mickmister Sure I will make it editable. If we want to attach a comment from message actions of slack attachment type post then the comment field remains empty.

Nityanand13 committed:
Merge branch 'master' of github.com:mattermost/mattermost-plugin-github into MM-618

Nityanand13 committed:
[MI-2736]: Done the review fixes of a github PR #636 (#20)

* [MI-2736]: Review fixes done
1. Improved code readability

* [MI-2736]: Review fixes done
1. Fixed linting errors

* [MI-2736]: Review fixes done
1. Fixed linting error

mickmister commented on server/plugin/api.go:
```diff
@@ -139,6 +95,11 @@ func (p *Plugin) initializeAPI() {
+	apiRouter.HandleFunc("/closeorreopenissue", p.checkAuth(p.attachUserContext(p.closeOrReopenIssue), ResponseTypePlain)).Methods(http.MethodPost)
+	apiRouter.HandleFunc("/updateissue", p.checkAuth(p.attachUserContext(p.updateIssue), ResponseTypePlain)).Methods(http.MethodPost)
+	apiRouter.HandleFunc("/editissuemodal", p.checkAuth(p.attachUserContext(p.openIssueEditModal), ResponseTypePlain)).Methods(http.MethodPost)
+	apiRouter.HandleFunc("/close_reopen_issue_modal", p.checkAuth(p.attachUserContext(p.openCloseOrReopenIssueModal), ResponseTypePlain)).Methods(http.MethodPost)
```
I meant to have `snake_case` applied to the other endpoints here too. I see we're already using `lazycase` on others so it doesn't matter I suppose. Good to be consistent though

Nityanand13 committed:
[MI-2736]: Done the review fixes of a github PR #636 (#21)

* [MI-2736]: Review fixes done
1. Improved code readability

* [MI-2736]: Review fixes done
1. Fixed linting errors

* [MI-2736]: Review fixes done
1. Fixed linting error

* [MI-2736]: Review fixes done
1. Improved code readability

* [MI-2736]: Review fixes done
1. Changed the case of few endpoints to snake case

mickmister commented on webapp/src/components/github_issue/index.tsx:
```diff
@@ -0,0 +1,118 @@
+};
+
+const getStyle = makeStyleFromTheme((theme) => ({
+    button: {
+        fontFamily: 'Open Sans',
+        fontSize: '12px',
+        fontWeight: 'bold',
+        letterSpacing: '1px',
+        lineHeight: '19px',
+        margin: '12px 12px 8px 0px',
+        borderRadius: '4px',
+        color: theme.buttonColor,
+    },
```
Instead of these styles, can we make the buttons match the styling of most other buttons in the MM UI? There are class names used for this, in the footer of the modals in this project for instance. The current styling seems inconsistent with the rest of the application

mickmister replied:
@DHaussermann I'm curious about your thoughts on this. Should these buttons be made to look like the slack attachment buttons as much as possible?

mickmister commented on server/plugin/webhook.go:
```diff
@@ -549,12 +539,35 @@ func (p *Plugin) postIssueEvent(event *github.IssuesEvent) {
 			Message: renderedMessage,
 		}
 
+		if action == constants.ActionOpened {
+			post.Type = "custom_git_issue"
+			post.Props = map[string]interface{}{
+				constants.TitleForProps:       *issue.Title,
+				constants.IssueURLForProps:    *issue.HTMLURL,
+				constants.IssueNumberForProps: *issue.Number,
+				constants.DescriptionForProps: description,
```
So the current conditions on this is to show these controls on a post created by the plugin, specifically for when an issue is opened.

I think it would be useful to have for PRs and when labeling issues. And possibly in the bot DMs as well, though that could get noisy. @hanzei What are your thoughts on this?

hanzei replied:
1/5 that it's handy to have actions attached to the DM notification as this allows user to answer them directly in MM. Maybe we even add the buttons to all posts?

@asatkinson What do you think?

mickmister reviewed (commented):
Just a few more questions from my side. I can approve once conversation is wrapped up

hanzei commented on server/constants/constants.go:
```diff
@@ -0,0 +1,46 @@
+package constants
```
From https://go.dev/blog/package-names

> Avoid meaningless package names. Packages named util, common, or misc provide clients with no sense of what the package contains.


What value does a separate package provide?

Nityanand13 replied:
@hanzei Keeping things in separate package helps us in writing unit tests and creating mocks for packages.

hanzei replied:
4/5 that a `constants` package is not an idiomatic way to store variables, as it contradicts the above quote.

hanzei commented on server/serializer/error.go:
```diff
@@ -0,0 +1,11 @@
+package serializer
```
Same question about the package structure here.

hanzei commented on server/plugin/api.go:
```diff
@@ -1035,70 +959,223 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
+		return
+	}
+
+	userID := r.Header.Get(constants.HeaderMattermostUserID)
```
Can `c.GHInfo.UserID` be used here instead?

hanzei commented on server/plugin/api.go:
```diff
@@ -1035,70 +959,223 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
+		return
+	}
+
+	p.API.PublishWebSocketEvent(
+		wsEventAttachCommentToIssue,
+		map[string]interface{}{
+			"postId": post.Id,
+			"owner":  req.RepoOwner,
+			"repo":   req.RepoName,
+			"number": req.IssueNumber,
+		},
+		&model.WebsocketBroadcast{UserId: userID},
+	)
```
Why is the request to the plugin needed if it simply returns a websocket event?

hanzei replied:
The same question goes for openCloseOrReopenIssueModal.

Nityanand13 replied:
Now we are not making any request for these cases

hanzei commented on server/plugin/api.go:
```diff
@@ -1035,70 +959,223 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
+		return
+	}
+
+	p.API.PublishWebSocketEvent(
+		wsEventCreateOrUpdateIssue,
+		map[string]interface{}{
+			"title":            *issue.Title,
+			"channel_id":       post.ChannelId,
+			"postId":           req.PostID,
+			"milestone_title":  milestoneTitle,
+			"milestone_number": milestoneNumber,
+			"assignees":        assignees,
+			"labels":           labels,
+			"description":      description,
+			"repo_full_name":   fmt.Sprintf("%s/%s", req.RepoOwner, req.RepoName),
+			"issue_number":     *issue.Number,
+		},
+		&model.WebsocketBroadcast{UserId: userID},
+	)
```
Why is a websocket event used and not a standard HTTP response?

Nityanand13 replied:
Now used HTTP response

mattermost-build commented:
This PR has been automatically labelled "stale" because it hasn't had recent activity.
A core team member will check in on the status of the PR to help with questions.
Thank you for your contribution!

Nityanand13 committed:
[MI-2814] Done the review fixes of github PR #636 (#22)

* [MI-2736]: Review fixes done
1. Improved code readability

* [MI-2736]: Review fixes done
1. Fixed linting errors

* [MI-2736]: Review fixes done
1. Fixed linting error

* [MI-2736]: Review fixes done
1. Improved code readability

* [MI-2736]: Review fixes done
1. Changed the case of few endpoints to snake case

* [MI-2814]: Review fixes done of github PR #636

* [MI-2814]: Review fixes done
1. Made comment field editable when trying to add a comment from message action.

* [MI-2814]: Review fixes done
1. Improved code readability

* [MI-2814]: Review fixes done
1. Used GET method on getting the issue info
2. Improved code readability

* [MI-2814]: Improved linting error

Nityanand13 committed:
Merge branch 'master' of github.com:mattermost/mattermost-plugin-github into MM-618

hanzei commented:
@Nityanand13 @Kshitij-Katiyar Could one of you please resolve the merge conflicts?

Kshitij-Katiyar commented:
> @Nityanand13 @Kshitij-Katiyar Could one of you please resolve the merge conflicts?

@hanzei sure

ayusht2810 committed:
Merge branch 'master' of github.com:mattermost/mattermost-plugin-github into MM-618

ayusht2810 committed:
[MM-618] Update validations, modal title, styling, post messages, conflicts error and lint errors

mickmister commented:
@asaadmahmood Can I get your opinion on this feature?

@ayusht2810 Are you able to get this running on a Cloud server so we can give access to others for review?

ayusht2810 committed:
Merge branch 'master' of github.com:mattermost/mattermost-plugin-github into MM-618

ayusht2810 commented:
> @ayusht2810 Are you able to get this running on a Cloud server so we can give access to others for review?

@mickmister Do you mean to create a cloud server and deploy the build for this PR? If yes, what should I use for client ID and client secret for OAuth?

mickmister commented:
@ayusht2810 You should be able to use the `Use Preregistered OAuth Application` feature here. Then `/github connect` should just work without any further configuration.

![image](https://github.com/mattermost/mattermost-plugin-github/assets/6913320/a2e5ad9d-43f0-4211-a807-42a19432ee49)

ayusht2810 commented:
@mickmister Created a cloud server for testing and review of the PR and configured the plugin on it as well. https://github-issues-test.test.mattermost.cloud

mickmister commented:
@asaadmahmood Would you be up for taking a look at this feature? @ayusht2810 can give you access to the cloud server if you want to review the feature in action

hanzei reviewed (changes requested):
Please fix the merge conflicts, thanks!

hanzei commented:
Happy to take a look afterwards

ayusht2810 committed:
Merge branch 'master' of github.com:mattermost/mattermost-plugin-github into MM-618

ayusht2810 committed:
[MM-618] Remove constants and serializers package

ayusht2810 commented:
@hanzei fixed the review fixes and resolved the conflicts.

mattermost-build commented:
This PR has been automatically labelled "stale" because it hasn't had recent activity.
A core team member will check in on the status of the PR to help with questions.
Thank you for your contribution!

mickmister commented:
@ayusht2810 I'd like to make sure the UX for this feature is approved before going forward with the PR. It seems off to me to have a destructive button ("Close" in this case) be so readily available and visible. That's typically not what I would expect to see in this context. Maybe we can have a "Edit" button, which can open a modal to give all the options? Are you able to provide more screenshots and/or gifs of the feature working?

ayusht2810 committed:
Merge branch 'master' of github.com:mattermost/mattermost-plugin-github into MM-618

ayusht2810 commented:
@mickmister Below are different screenshots for the feature. Please let me know if anything else is needed:
- Issue creation post
![image](https://github.com/mattermost/mattermost-plugin-github/assets/100013900/6d55e46d-f98f-4be6-8d3a-0dc821f38e76)
- Create comment modal
![image](https://github.com/mattermost/mattermost-plugin-github/assets/100013900/886291d5-0ad1-4e87-84b5-743e2faccd21)
- Comment added post
![image](https://github.com/mattermost/mattermost-plugin-github/assets/100013900/9db56d6f-6ce0-4f87-9871-8c028379218f)
- Update issue modal
![image](https://github.com/mattermost/mattermost-plugin-github/assets/100013900/ff444e3f-1fa2-4858-b270-4abb021c8863)
- Issue updated added post
![image](https://github.com/mattermost/mattermost-plugin-github/assets/100013900/8d763890-6858-4ec1-aa87-2c66c3f80f63)
- Updated post after after updating issue (same post as previous)
![image](https://github.com/mattermost/mattermost-plugin-github/assets/100013900/dbabe8e0-bb06-49eb-93d5-b5cc5c5e7ed8)
- Close issue modal
![image](https://github.com/mattermost/mattermost-plugin-github/assets/100013900/99397669-6a1e-467d-af36-3dec3d8104e3)
- Updated post after closing issue
![image](https://github.com/mattermost/mattermost-plugin-github/assets/100013900/68ba3490-fe80-4149-9f7b-fac30f0ca06a)
- Reopen modal
![image](https://github.com/mattermost/mattermost-plugin-github/assets/100013900/8d93c571-586b-4ed9-872d-7e57b047f070)

mickmister commented:
@matthewbirtch This feature was suggested by a previous PM on the integrations, to essentially mirror slack's integration that has this functionality https://github.com/mattermost/mattermost-plugin-github/issues/618

The main thing I'm weary of in this feature is that we are surfacing a `Close` button on the post, when that's not an action a user would typically want to take within Mattermost. Personally I think we should remove the `Close` button. Do you have any thoughts on this?

matthewbirtch commented:
> @matthewbirtch This feature was suggested by a previous PM on the integrations, to essentially mirror slack's integration that has this functionality #618
> 
> The main thing I'm weary of in this feature is that we are surfacing a `Close` button on the post, when that's not an action a user would typically want to take within Mattermost. Personally I think we should remove the `Close` button. Do you have any thoughts on this?

@mickmister I don't have a problem with the `Close` button, to be honest. Because it launches a modal, it's not really prone to misclicks. If someone thinks they have enough context within Mattermost to be able to close an issue, I don't have a big problem with giving them that ability. What are your specific concerns?

I have some unrelated thoughts about tweaks to styling, but I'll reserve that until it's ready for that.

mickmister commented:
>  I don't have a problem with the `Close` button, to be honest. Because it launches a modal, it's not really prone to misclicks. If someone thinks they have enough context within Mattermost to be able to close an issue, I don't have a big problem with giving them that ability. What are your specific concerns?

@matthewbirtch That all makes sense. My main concern is the "danger" color in the channel. These posts are mostly meant to be informational, and user actions will mostly not be taken. Having the danger passively "at your fingertips" seems alarming to me. I think the likelihood of closing is much lower than commenting and editing as well.

matthewbirtch commented:
> @matthewbirtch That all makes sense. My main concern is the "danger" color in the channel. These posts are mostly meant to be informational, and user actions will mostly not be taken. Having the danger passively "at your fingertips" seems alarming to me. I think the likelihood of closing is much lower than commenting and editing as well.

Fair points @mickmister. However, the purpose of this PR is to introduce more interactivity to the post to allow users to take action. This could reduce the need to switch out to GitHub and remain in Mattermost. If we're allowing editing or commenting, why not closing? We could remove the danger color though - that might make it less alarming. I'd be ok with that since it's not immediately destructive (due to the modal that opens) and would reduce the prominance it creates.

matthewbirtch commented:
Is it possible to get this up on a test server to play with as well? The button styles don't look quite right to me either.

Commit:
Merge branch 'master' of github.com:mattermost/mattermost-plugin-github into MM-618

raghavaggarwal2308 commented:
@matthewbirtch Please refer this message: https://hub.mattermost.com/partners/pl/bdbujx3ebibuumiifmymwtuj8c

matthewbirtch commented on webapp/src/components/github_issue/index.tsx:
```diff
@@ -0,0 +1,119 @@
+
+const getStyle = makeStyleFromTheme((theme) => ({
+    button: {
+        fontFamily: 'Open Sans',
```
You should be able to remove all of these styles since you're already using the `btn btn-primary` classes

matthewbirtch replied:
In addition, we don't need to make the `close` button red since it's not immediately destructive

matthewbirtch commented on webapp/src/components/github_issue/index.tsx:
```diff
@@ -0,0 +1,119 @@
+
+    return (
+        <div>
+            <h5>
```
I'd like to make the title `<h4>` and change `font-family: Metropolis` and `font-weight: 600`

matthewbirtch commented on webapp/src/components/github_issue/index.tsx:
```diff
@@ -0,0 +1,119 @@
+                    {'#' + postProps.issue_number + ' ' + postProps.title}
+                </a>
+            </h5>
+            <p>{postProps.description}</p>
```
It seems that this paragraph is missing a bottom margin and ends up sitting too close to the assignees and labels

Commit:
Merge branch 'master' of github.com:mattermost/mattermost-plugin-github into MM-618

Commit:
[MM-556] Review fixes

raghavaggarwal2308 commented:
@matthewbirtch Updated the post acc to your suggestion. Please re-review. 
![image](https://github.com/mattermost/mattermost-plugin-github/assets/55234496/b3c340d6-65b6-48ec-bfa5-8ec782d0daa4)

matthewbirtch commented:
> @matthewbirtch Updated the post acc to your suggestion. Please re-review. ![image](https://private-user-images.githubusercontent.com/55234496/346086658-b3c340d6-65b6-48ec-bfa5-8ec782d0daa4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAxOTUyNDYsIm5iZiI6MTcyMDE5NDk0NiwicGF0aCI6Ii81NTIzNDQ5Ni8zNDYwODY2NTgtYjNjMzQwZDYtNjViNi00OGVjLWJmYTUtOGVjNzgyZDBkYWE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA1VDE1NTU0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJhNWQwMTVlYWNhOTUwNzNkYjBlNjAxZGM1ZjcxMWNmOGRhNDU4MWJkMDg3NTAyZGY2MjM2ZjYwMzM3YTY2MGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OAVRHgIQGe0OrytlcOlacKFhsSlFbO9-dad72p9aBjc)

Thanks @raghavaggarwal2308, the screenshot looks good, but could you update the test server? These changes aren't showing for me there.

raghavaggarwal2308 commented:
> Thanks @raghavaggarwal2308, the screenshot looks good, but could you update the test server? These changes aren't showing for me there.

@matthewbirtch The test server is updated now.

matthewbirtch commented:
Thanks @raghavaggarwal2308. A few tiny details:
1. Since this post shows when a new issue is created, I wonder if we need a line at the beginning of these posts that says '{User} created a new issue in {owner/repo}'. Otherwise it doesn't really provide a reason why this is getting posted in the channel.

2. I'm also thinking that the details of the issue should show within a card UI element.
    ![image](https://github.com/mattermost/mattermost-plugin-github/assets/2040554/5ef81089-3fb3-4cef-8ada-26b50a9a604e)


1. I'm seeing extra space above the buttons if there are no assignee/label fields showing. See comparison below:
    <img width="674" alt="Screenshot 2024-07-07 at 5 46 46 PM" src="https://github.com/mattermost/mattermost-plugin-github/assets/2040554/d65f0f96-5777-454f-981c-8bc45650071f">

4. Also, I'm wondering why we show the issue number first in the title, but in the hover popover, we show the issue number after the name and not bolded. I think we should be consistent here and show the issue number after the title.
    <img width="396" alt="image" src="https://github.com/mattermost/mattermost-plugin-github/assets/2040554/54bcf022-af10-4c19-b98f-b8df29c88acc">

5. On the mobile app, I'm seeing extra space below the issue title and the title is not linked like it is on the webapp. The title is also inconsistent with webapp as it doesn't include the issue number as part of the title. In addition, I think there should be a paragraph space here above the labels/assignee fields. Also noticed these fields are appearing in a different order than on the webapp.
    <img src="https://github.com/mattermost/mattermost-plugin-github/assets/2040554/a79697ff-a15a-4f3a-bb6c-c64f3c9e7ccc" width="375">

    
6. When I update an issue from the edit modal, a reply is posted that says `Updated GitHub issue #80 from a message`. I think we can remove 'from a message' as it seems odd in this context.
    <img width="496" alt="image" src="https://github.com/mattermost/mattermost-plugin-github/assets/2040554/e701ccbf-a296-434c-8cf2-a43dd3845e3d">

raghavaggarwal2308 commented:
@matthewbirtch All these suggestions will take some time to fix. Also, we will have to handle styling of many things. What are your opinions on replacing the custom post with a slack attachment for webapp as well?

matthewbirtch commented:
> @matthewbirtch All these suggestions will take some time to fix. Also, we will have to handle styling of many things:
> 
> 1. Adding the username and org link in the pretext. This will become quire easy in a slack attachment.
> 2. Rendering as a card.
> 
> . What are your opinions on replacing the custom post with a slack attachment for webapp as well?

I think that could work quite well and makes a lot of sense to me. @mickmister any objections to this?

mickmister commented:
@raghavaggarwal2308 Sounds good to me :+1:

Kshitij-Katiyar committed:
[MM-617]: converted the custom post to slack attachment for issue creation event on Github (#37)

* [MM-617]: converted the custom post to slack attachment for issue creation

* [MM-617]: fixed lint and testcases

* [MM-617]: removed unused variables

* [MM-617]: Fixed the title of the issue slackAttachment

* [MM-617]: review fixes

* [MM-617]: fixed lint

* [MM-617]: Review fixes and code clean up

* [MM-617]: removed the custom post file

* [MM-617]: review fixes

* [MM-617]: review fixes

* [MM-617]: Fixed lint

* [MM-617]: improved the error logging

Kshitij-Katiyar committed:
Merge branch 'master' of github.com:mattermost/mattermost-plugin-github into MM-618

raghavaggarwal2308 commented:
@matthewbirtch Converted the custom post to slack attachment and updated the screenshot in description. The test server is also updated with the latest build. Please re-review.

Kshitij-Katiyar committed:
Merge branch 'master' of github.com:mattermost/mattermost-plugin-github into MM-618

Kshitij-Katiyar committed:
[MM-618]: Fixed the lint

matthewbirtch commented:
@raghavaggarwal2308 the webapp looks good to me, but this is not rendering at all on mobile for me on the test server.

| Webapp | Mobile |
|--|--|
| <img width="502" alt="Screenshot 2024-09-03 at 11 04 40 AM" src="https://github.com/user-attachments/assets/0365f31f-097f-4617-9f55-01a159c5c560"> | <img width="400" src="https://github.com/user-attachments/assets/5a01503a-df26-4883-b43b-72d07ec394e3"> |

Kshitij-Katiyar commented:
> @raghavaggarwal2308 the webapp looks good to me, but this is not rendering at all on mobile for me on the test server.

@matthewbirtch I tried reproducing the issue but it works fine on my local server and the [maintenance testing cloud server](https://plugin-maintenance.test.mattermost.cloud).
If you can still reproduce the issue, please let us know the steps.

matthewbirtch commented:
@raghavaggarwal2308 it seems better now. Although, now I'm seeing a slightly different issue. 

I'm testing on your test server in the `Town Square` channel in the `devsecops` team and I'm still seeing some inconsistencies with the webapp.

1. The mobile app is showing two empty posts from the github bot. While the webapp does not show these.

| webapp | mobile app |
|--|--|
| <img width="1253" alt="image" src="https://github.com/user-attachments/assets/b1aa685d-a06f-48f7-96c5-ab0a5147f325"> | <img src="https://github.com/user-attachments/assets/e78b79e3-f4e0-4926-8b77-37fbeaf4c884" width="414"> |

2. Also, the mobile app is not showing the replies that I'm seeing on the webapp. However, when I tap on the post, it opens the thread and shows the replies. When I go back to the channel from there, it's showing the replies. I'm not sure if this is a bigger problem or if it's related to this PR. Here's a screen recording:

https://github.com/user-attachments/assets/2e05622c-442c-4caf-9d5a-7fc3a792b5ac

raghavaggarwal2308 commented:
@matthewbirtch Sure, we will look into it.

Kshitij-Katiyar commented:
> @raghavaggarwal2308 it seems better now. Although, now I'm seeing a slightly different issue.
@matthewbirtch 
I investigated the issues you reported and would like to share my findings:
- Regarding the blank post from the GitHub bot, I was unable to reproduce the issue. Could you try reproducing it in a new channel and team to ensure it's not caused by a custom component from the previous version of the PR?
- For the issue with replies to Slack attachments, I was able to reproduce it. After reviewing the PR, I found no missing `rootID` or related technical issue, suggesting the problem may lie on the Mattermost side.
Let me know your thoughts.

matthewbirtch commented:
> * Regarding the blank post from the GitHub bot, I was unable to reproduce the issue. Could you try reproducing it in a new channel and team to ensure it's not caused by a custom component from the previous version of the PR?

I think this is the case. It's probably a leftover from the previous implementation of the custom component. I think we're good with this part.

> For the issue with replies to Slack attachments, I was able to reproduce it. After reviewing the PR, I found no missing rootID or related technical issue, suggesting the problem may lie on the Mattermost side.
Let me know your thoughts.

For this one, I'll need support from @wiggin77 or @mickmister to identify if this is a plugin issue or a platform issue.

hanzei reviewed (changes requested):
Can you please resolve the conflicts first?

Kshitij-Katiyar committed:
Merge branch 'master' of github.com:mattermost/mattermost-plugin-github into MM-618

hanzei reviewed (commented):
@wiggin77 Can you please confirm from a product owner perspective that this is a feature we want to incorporate into the plugin?

wiggin77 commented:
> @wiggin77 Can you please confirm from a product owner perspective that this is a feature we want to incorporate into the plugin?

I will defer to @matthewbirtch as to whether this should be added.  If he approves the merge conflicts will be resolved and the PR tested.

wiggin77 commented:
> @wiggin77 Can you please confirm from a product owner perspective that this is a feature we want to incorporate into the plugin?

@hanzei this is approved by Eric.

matthewbirtch commented:
> > @wiggin77 Can you please confirm from a product owner perspective that this is a feature we want to incorporate into the plugin?
> 
> I will defer to @matthewbirtch as to whether this should be added. If he approves the merge conflicts will be resolved and the PR tested.

Sorry for the delayed reply here. I'm aligned to this being added to the plugin, but I am unclear if the identified issues were resolved. Sorry, I may have lost the testing server to test this out though.

raghavaggarwal2308 commented:
> Also, the mobile app is not showing the replies that I'm seeing on the webapp. However, when I tap on the post, it opens the thread and shows the replies. When I go back to the channel from there, it's showing the replies. I'm not sure if this is a bigger problem or if it's related to this PR. Here's a screen recording:

@matthewbirtch There is only one outstanding issue. Which according to our understanding is related to Mattermost server and not to the plugin code.
https://github.com/mattermost/mattermost-plugin-github/pull/636#issuecomment-2328931642

cc: @wiggin77

Kshitij-Katiyar committed:
Merge branch 'master' of github.com:mattermost/mattermost-plugin-github into MM-618

Kshitij-Katiyar committed:
fxed testcases

Kshitij-Katiyar committed:
merge master to the PR

Kshitij-Katiyar committed:
fixed lint

Kshitij-Katiyar commented:
> 2. Also, the mobile app is not showing the replies that I'm seeing on the webapp. However, when I tap on the post, it opens the thread and shows the replies. When I go back to the channel from there, it's showing the replies. I'm not sure if this is a bigger problem or if it's related to this PR. Here's a screen recording:

@matthewbirtch The replies on the mobile app are synced with the webapp and are now visible from outside the thread also

![Screenshot from 2025-01-31 18-47-11](https://github.com/user-attachments/assets/12dafabb-05de-40b0-80fc-69dc918a32e6)
![rn_image_picker_lib_temp_44d6fb50-511f-4ad0-acd8-66aeeac55b5e](https://github.com/user-attachments/assets/7e761ac2-35ea-4434-bbf4-afb48144f1a7)

Kshitij-Katiyar commented:
@AayushChaudhary0001 Please test this PR

matthewbirtch commented:
> Here is a [cloud server](https://plugin-maintenance.test.mattermost.cloud/qa/channels/github-testing) with the latest build of this PR uploaded in case you want to retest.

Thanks @Kshitij-Katiyar. The channel you linked to doesn't have any github posts. Is there another channel for me to test this, or can we get some more test posts from the github bot in this channel to test with?

Kshitij-Katiyar commented:
> > Here is a [cloud server](https://plugin-maintenance.test.mattermost.cloud/qa/channels/github-testing) with the latest build of this PR uploaded in case you want to retest.
> 
> Thanks @Kshitij-Katiyar. The channel you linked to doesn't have any github posts. Is there another channel for me to test this, or can we get some more test posts from the github bot in this channel to test with?

@matthewbirtch Added a post in the [server](https://plugin-maintenance.test.mattermost.cloud/qa/channels/github-testing). Do let me know if it is not visible to you

matthewbirtch commented:
Thanks @Kshitij-Katiyar. It looks good, however the buttons don't seem to work for me. The don't open modals on mobile.

Strangely, if I had the browser open on my desktop in the same channel and when I tapped the `Comment` or `Edit` buttons on mobile, it opened the corresponding modals on the desktop browser. But nothing happens on the mobile app.

wiggin77 commented:
> @AayushChaudhary0001 Please test this PR

Did this get tested @AayushChaudhary0001 ?

raghavaggarwal2308 commented:
> Did this get tested @AayushChaudhary0001 ?

@wiggin77 Not yet, there is an issue as mentioned by @matthewbirtch above. Once that is fixed we can proceed with the QA review

mattermost-build commented:
This PR has been automatically labelled "stale" because it hasn't had recent activity.
A core team member will check in on the status of the PR to help with questions.
Thank you for your contribution!

wiggin77 commented:
Let me know when this is ready for review.

Kshitij-Katiyar committed:
merge master to bs/MM-618

Kshitij-Katiyar committed:
Merge branch 'master' of github.com:mattermost/mattermost-plugin-github into MM-618

Kshitij-Katiyar committed:
updated snapshot

Kshitij-Katiyar committed:
fixed lint

Kshitij-Katiyar committed:
convert comment custom post to interactive dialog

Kshitij-Katiyar committed:
convert close/reopen issue custom modal to interactive dialog

Kshitij-Katiyar committed:
convert edit issue custom modal to interactive dialog

Kshitij-Katiyar commented:
Here are the screenshots after updating the custom modals to interactive dialog
![Screenshot from 2025-04-29 19-48-49](https://github.com/user-attachments/assets/f16128ab-5455-4e1f-b9a7-6cea47592eba)
![Screenshot from 2025-04-29 19-48-57](https://github.com/user-attachments/assets/721426a4-18f3-4f15-bb39-128db088e687)
![Screenshot from 2025-04-29 19-49-07](https://github.com/user-attachments/assets/476c16e5-946d-4014-a158-970c8397ff6e)
![Screenshot from 2025-04-29 19-49-18](https://github.com/user-attachments/assets/db7b26d2-d3bd-402c-9b7c-3b3f2a3aea65)
![Screenshot from 2025-04-29 19-49-32](https://github.com/user-attachments/assets/dfd6a0a0-ead6-47f8-a2fa-d41fc125f627)
![Screenshot from 2025-04-29 19-49-43](https://github.com/user-attachments/assets/4340ffd4-33bf-4eb9-9995-e94349b2fad2)

Kshitij-Katiyar commented:
Here are the screenshots from mobile app
![1000334953](https://github.com/user-attachments/assets/7dadf11a-89fb-4b57-ac0b-ec198a2c44a7)
![1000334954](https://github.com/user-attachments/assets/9da4c0cd-b965-4a30-8698-09232a187241)
![1000334956](https://github.com/user-attachments/assets/392aff82-b33e-4d5c-ac2d-97c184c705b2)
![1000334955](https://github.com/user-attachments/assets/1000aeac-2e11-4065-85d8-860bb6635c79)

Kshitij-Katiyar committed:
Lint fixes

wiggin77 reviewed (changes requested):
@Kshitij-Katiyar removing from my queue until issues raised by reviewers are addressed and merge conflicts resolved.

abbas-dependable-naqvi commented:
Can be taken forward when multi select dropdown is supported by Mattermost in interactive dialogs
